apps/test/unit/templates/sectionProgress/progressTables/ProgressTableDetailCellTest.jsx

```diff
@@ -0,0 +1,177 @@
+  });
+
+  it('disables progress bubble if there is a bonus and stageExtrasEnabled is false', () => {
+    const wrapper = shallow(<ProgressTableDetailCell {...DEFAULT_PROPS} />);
+    const levelBubble3 = wrapper.findWhere(node => node.key() === '999_3');
+    const progressBubble = levelBubble3.find(ProgressTableLevelBubble);
+    expect(progressBubble.props().disabled).to.equal(true);
+  });
+
+  it('disable is false for progress bubble if there is a bonus and stageExtrasEnabled is true', () => {
+    const props = {...DEFAULT_PROPS, stageExtrasEnabled: true};
+    const wrapper = shallow(<ProgressTableDetailCell {...props} />);
+    const levelBubble3 = wrapper.findWhere(node => node.key() === '999_3');
+    const progressBubble = levelBubble3.find(ProgressTableLevelBubble);
+    expect(progressBubble.props().disabled).to.equal(false);
+  });
+
+  it('generates the right url for level bubble', () => {
+    const wrapper = shallow(<ProgressTableDetailCell {...DEFAULT_PROPS} />);
+    const levelBubble1 = wrapper.findWhere(node => node.key() === '123_1');
+    const url = levelBubble1.find(ProgressTableLevelBubble).props().url;
+    expect(url).to.equal('/level-1?section_id=123&user_id=1');
+  });
+
+  it('calls firehouse putRecord when clicking a level', () => {
+    const wrapper = shallow(<ProgressTableDetailCell {...DEFAULT_PROPS} />);
+    const levelBubble1 = wrapper
+      .findWhere(node => node.key() === '123_1')
+      .childAt(0);
+    levelBubble1.simulate('click');
+    expect(firehoseClient.putRecord).to.have.been.called;
+  });
+
+  it('calls firehouse putRecord when clicking a sublevel', () => {
+    const wrapper = shallow(<ProgressTableDetailCell {...DEFAULT_PROPS} />);
+    const sublevel = wrapper.findWhere(
+      node => node.key() === `${SUBLEVEL_1.id}`
+    );
+    sublevel.simulate('click');
+    expect(firehoseClient.putRecord).to.have.been.called;
+  });
+
+  it('widthForLevels returns the correct width when there are unplugged levels', () => {
+    const levels = [LEVEL_1, LEVEL_1];
+    const unpluggedWidth = unitTestExports.getUnpluggedWidth();
+    const expectedWidth = 2 * unpluggedWidth + 2 * progressStyles.CELL_PADDING;
+    expect(ProgressTableDetailCell.widthForLevels(levels)).to.equal(
+      expectedWidth
+    );
+  });
+
+  it('getUnpluggedWidth returns the width of the unplugged bubble in english', () => {
+    expect(unitTestExports.getUnpluggedWidth()).to.equal(121);
+  });
+
+  it('getUnpluggedWidth returns the width of the unplugged bubble in spanish', () => {
+    sinon.stub(i18n, 'unpluggedActivity').returns('Actividad fuera de línea');
+    expect(unitTestExports.getUnpluggedWidth()).to.equal(147);
+  });
+
+  it('widthForLevels returns the correct width when there are no unplugged levels', () => {
+    const notUnpluggedLevel = {...LEVEL_1, isUnplugged: false};
+    const levels = [notUnpluggedLevel, notUnpluggedLevel];
+
+    const unpluggedLevelCount = 2;
```

Comment:
i think this is a copy/paste typo. reading "there are no unplugged levels.... unpluggedLevelCount = 2" kinda tripped me up.